### PR TITLE
bank: Added user to user transfer

### LINKED ISSUE #51: bank: transactions between users
The bank functionality must be expanded, so that users would be able to transfer funds between each other and to different services such as taxing, paying the bills, online shopping and etc. This will provide new functionalities to the users, thus increasing the traffic to our application.

### PR DESCRIPTION
**Pull request by gadjevb**
Added user to user transfer. Now users have the ability to transfer money between their accounts.

Fixes #51 

screenshot from 2017-02-09 14-53-23

### CODE REVIEW

**Comment by gadjevb** [comment]
@mgenov , @martinmilev ready for review!

File: src/main/java/com/clouway/bank/adapter/http/TransferService.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    this.accountRepository = accountRepository;
+    this.transferRepository = transferRepository;
+  }
+
+  @Get
+  public Reply<?> returnCurrentUser() {
+    Optional<User> possibleUser = security.currentUser();
+    Optional<Account> possibleAccount;
+
+    if (possibleUser.isPresent()) {
```
Comment (mgenov) [inline src/main/java/com/clouway/bank/adapter/http/TransferService.java]: The design smells here as you are doing one check for the user and one for the account. 

UserSecurity could return User directly.

File: src/main/java/com/clouway/bank/adapter/persistence/PersistentTransferRepository.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    transfers().insertOne(document);
+  }
+
+  @Override
+  public List<TransferRecord> retrieveTransfers(String userID, String startingFromCursor, Boolean isNext, Integer limit) {
+    List<TransferRecord> records = new ArrayList();
+
+    MongoCursor<Document> docCursor;
+
+    if(startingFromCursor.equals("")) {
```
Comment (mgenov) [inline src/main/java/com/clouway/bank/adapter/persistence/PersistentTransferRepository.java]: You are duplicating searching and you can consider extracting it as separate object or at least you can add comment about this duplication.

File: src/main/java/com/clouway/bank/adapter/http/TransferService.java [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+      return Reply.saying().badRequest();
+
+    }
+
+    if (sender.isPresent() && receiver.isPresent() && (senderAcc.get().balance >= query.amount)) {
+
+      Double senderNewBalance = senderAcc.get().balance - query.amount;
+      Double receiverNewBalance = receiverAcc.get().balance + query.amount;
+
+      transferRepository.saveTransfer(new Transfer(
```
Comment (mgenov) [inline src/main/java/com/clouway/bank/adapter/http/TransferService.java]: You need to add some comment that this operation is not safe as update could fail and there is no retry logic.

**Comment by gadjevb** [comment]
@mgenov commented the duplicated code and the not save saveTransfer() method.

**Pull request closed without merging**